### PR TITLE
feat: Modify plugin preview page

### LINKED ISSUE #1991: 在预览插件的时候，建议只显示启动的插件而不是显示所有的插件
## Issue description
在预览插件的时候显示了所有的插件
image

## Expected behavior
建议只显示启动的插件而不是显示所有的插件
image

## Environment

* apisix version (cmd: `apisix version`): 2.7
* apisix-dashboard version, if have: 2.7
* Browser version, if have:

### PR DESCRIPTION
**Pull request by Si-ege**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #1991
1.The preview page shows only the activated plug-ins.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: web/src/components/Plugin/PluginPage.tsx [hunk 1]
```diff
@@ -179,66 +183,131 @@ const PluginPage: React.FC<Props> = ({
...
-                      type={
-                        initialData[item.name] && !initialData[item.name].disable
-                          ? 'primary'
-                          : 'default'
-                      }
-                      danger={initialData[item.name] && !initialData[item.name].disable}
-                      onClick={() => {
-                        setName(item.name);
+        {readonly
+          ? new_openPluginType.map((typeItem) => {
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:187]: We can update dom automatically by changing the source data 😄

File: web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js [hunk 2]
```diff
@@ -74,6 +75,8 @@ context('Create Configure and Delete PluginTemplate', () => {
 
     cy.contains('Submit').click();
     cy.contains('Next').click();
+    cy.contains(selector.pluginCard, 'basic-auth').should('be.visible');
+    cy.contains(selector.pluginTitle, 'Authentication').should('be.visible');
```
Comment (guoqqqi) [inline web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js:79]: We'd better add assertions to determine that other category plugins don't exist

File: web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js [hunk 3]
```diff
@@ -88,6 +91,8 @@ context('Create Configure and Delete PluginTemplate', () => {
     cy.get(selector.description).clear().type(data.pluginTemplateName2);
     cy.contains('Next').click();
     cy.contains('Next').click();
+    cy.contains(selector.pluginCard, 'basic-auth').should('be.visible');
+    cy.contains(selector.pluginTitle, 'Authentication').should('be.visible');
```
Comment (guoqqqi) [inline web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js:95]: ditto

File: web/src/components/Plugin/PluginPage.tsx [hunk 4]
```diff
@@ -92,6 +92,14 @@ const PluginPage: React.FC<Props> = ({
       form.setFieldsValue({ plugin_config_id });
     });
   }, []);
+  console.log('initialData: ', initialData);
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:95]: please remove this console

**Comment by guoqqqi** [comment]
Hi, @Si-ege can you add some images for this PR?

**Comment by Si-ege** [comment]
image

File: web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js [hunk 5]
```diff
@@ -74,6 +75,17 @@ context('Create Configure and Delete PluginTemplate', () => {
...
+    cy.contains(
+      selector.pluginTitle,
+      'Security',
+      'Traffic Control',
+      'Serverless',
+      'Observability',
+      'Other',
```
Comment (guoqqqi) [inline web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js:86]: I'm not sure this is the right way to write them, should we split them into 5 assertions

**Comment by guoqqqi** [comment]
also cc @Baoyuantop PTAL

File: web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js (hunk 2 above)
Comment (Baoyuantop) [reply web/cypress/integration/pluginTemplate/create-edit-delete-plugin-template.spec.js:79]: not exist or not visible?

File: web/src/components/Plugin/PluginPage.tsx [hunk 6]
```diff
@@ -92,6 +92,13 @@ const PluginPage: React.FC<Props> = ({
       form.setFieldsValue({ plugin_config_id });
     });
   }, []);
+  const openPlugin = pluginList.filter(
+    (item) => initialData[item.name] && !initialData[item.name].disable,
+  );
+  const openPluginType = openPlugin.map((item) => item.type);
+  const newOpenPluginType = openPluginType.filter((elem, index, self) => {
+    return index === self.indexOf(elem);
+  });
```
Comment (Baoyuantop) [inline web/src/components/Plugin/PluginPage.tsx:101]: Can you explain it? I'm a little confused.

**Review by Baoyuantop — COMMENTED** [review]
nice work~

File: web/src/components/Plugin/PluginPage.tsx [hunk 7]
```diff
@@ -92,6 +92,13 @@ const PluginPage: React.FC<Props> = ({
       form.setFieldsValue({ plugin_config_id });
     });
   }, []);
+  const openPlugin = pluginList.filter(
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:95]: `openPlugin` is Not a good variable name, It should be a list, for example: `openPluginList`

**Review by guoqqqi — COMMENTED** [review]
other LGTM, also cc @hanzhenfang take a look

**Review by guoqqqi — APPROVED** [review]
LGTM cc @Baoyuantop @SkyeYoung

**Review by Baoyuantop — APPROVED** [review]

**Comment by guoqqqi** [comment]
image
image
Cool 👍

**Review by SkyeYoung — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**